trieste/models/keras/architectures.py

```diff
@@ -233,3 +235,148 @@ def connect_layers(self) -> tuple[tf.Tensor, tf.Tensor]:
 
         return input_tensor, output_tensor
+
+
+class DropoutNetwork(KerasEnsembleNetwork):
+    """
+    This class builds a standard dropout neural network using Keras. The network
+    architecture is a multilayer fully-connected feed-forward network, with Dropout layers
+    preceding each fully connected dense layer. The network is meant to be passed to 
+    :class:`MCDropout` which will define the predict method to make this a probabilistic model.
+    """
+
+    def __init__(
+        self,
+        input_tensor_spec: tf.TensorSpec,
+        output_tensor_spec: tf.TensorSpec,
+        hidden_layer_args: Sequence[dict[str, Any]] = (
+            {"units": 50, "activation": "relu"},
+            {"units": 50, "activation": "relu"},
+            {"units": 50, "activation": "relu"},
+        ),
+        rate: Sequence[float] | float = 0.5
+    ):
+        """
+        :param input_tensor_spec: Tensor specification for the input to the network.
+        :param output_tensor_spec: Tensor specification for the output of the network.
+        :param hidden_layer_args: Specification for building dense hidden layers. Each element in
+            the sequence should be a dictionary containing arguments (keys) and their values for a
+            :class:`~tf.keras.layers.Dense` hidden layer. Please check Keras Dense layer API for
+            available arguments. Objects in the sequence will sequentially be used to add
+            :class:`~tf.keras.layers.Dense` layers. Length of this sequence determines the number of
+            hidden layers in the network. Default value is two hidden layers, 50 nodes each, with
+            ReLu activation functions. Empty sequence needs to be passed to have no hidden layers.
+        :param rate: Probability of dropout assigned to each layer of a fully connected network.
+            If scalar, the same probability will be assigned to each layer. Otherwise accepts a list of 
+            probabilities corresponding to each layer. 
+        :raise ValueError: If objects in ``hidden_layer_args`` are not dictionaries.
+        """
+        super().__init__(input_tensor_spec, output_tensor_spec)
+
+        self._hidden_layer_args = hidden_layer_args
+        self.rate = rate
+        self._model = self._build_model()
+
+    @property
+    def model(self) -> tf.keras.Model:
+        return self._model
+
+    @property
+    def rate(self) -> Sequence[float]:
+        return self._rate
+    
+    @rate.setter
+    def rate(self, rate):
+        if isinstance(rate, Sequence):
+            if not len(rate) == len(self._hidden_layer_args) + 1:
+                raise ValueError(
+                    f"""Requires a dropout probability for each hidden layer and the output layer. 
+                    Got {len(rate)} dropout probabilities for {len(self._hidden_layer_args) + 1} layers."""
+                )
+            for p in rate:
+                self._check_probability(p)
```

Comment:
check `tf.debugging` you should be able to do this check with functions there, no need for this `_check_probability` thing

Reply:
I tried to take a look at the tf.debugging module, but I couldn't find anything of the sort? There are regular assert greater than or equal to's. I also looked at the tensorflow Dropout layer and there they just use a similar raise ValueError check.